miner/src/transaction_queue.rs

```diff
@@ -86,14 +86,30 @@
 
 use std::default::Default;
 use std::cmp::{Ordering};
-use std::collections::{HashMap, BTreeSet};
+use std::collections::{HashMap, HashSet, BTreeSet};
 use util::numbers::{Uint, U256};
 use util::hash::{Address, H256};
 use util::table::*;
 use ethcore::transaction::*;
 use ethcore::error::{Error, TransactionError};
-use super::TransactionImportResult;
 
+macro_rules! check_too_cheap {
```

Comment:
Why not a regular function returning `Result<(), TransactionError >` instead of a macro?
`try!(check_limit(...))`